pkg/pget.go

```diff
@@ -25,14 +26,24 @@ type Options struct {
 }
 
 type ManifestEntry struct {
-	URL  string
-	Dest string
+	parsedURL *netUrl.URL
+	Dest      string
 }
 
-// A Manifest is a mapping from a base URI (consisting of scheme://authority) to
-// a list of Manifest entries under that base URI.  That is, the Manifest
-// entries are grouped by remote server that we might connect to.
-type Manifest map[string][]ManifestEntry
+func (m ManifestEntry) URL() string {
+	return m.parsedURL.String()
+}
+
+// A Manifest is a slice of ManifestEntry, with a helper method to add entries
+type Manifest []ManifestEntry
+
+func (m Manifest) AddEntry(url string, destination string) (Manifest, error) {
+	parsed, err := netUrl.Parse(url)
```

Comment:
why does AddEntry do the parsing? This feels like a bad abstraction.

Reply:
also, the only thing we do with this URL is call String() on it so why do we need to parse it at all? Why did we change it from string?